restcomm/restcomm.interpreter/src/main/java/org/mobicents/servlet/restcomm/interpreter/VoiceInterpreter.java

```diff
@@ -1147,24 +1147,53 @@ private void conferenceStateModeratorPresent(final Object message) {
             // https://telestax.atlassian.net/browse/RESTCOMM-710
             final SipServletRequest invite = callInfo.invite();
             // For outbound calls created with Calls REST API, the invite at this point will be null
-            if (invite != null)
-                processCustomHeaders(invite, "SipHeader_", parameters);
+            if (invite != null) {
+                processCustomAndDiversionHeaders(invite, "SipHeader_", parameters);
+
+            }
         } else {
-            processCustomHeaders(lastResponse, "SipHeader_", parameters);
+            processCustomAndDiversionHeaders(lastResponse, "SipHeader_", parameters);
         }
 
         return parameters;
     }
 
-    private void processCustomHeaders(SipServletMessage sipMessage, String prefix, List<NameValuePair> parameters) {
+    private void processCustomAndDiversionHeaders(SipServletMessage sipMessage, String prefix, List<NameValuePair> parameters) {
         Iterator<String> headerNames = sipMessage.getHeaderNames();
         while (headerNames.hasNext()) {
             String headerName = headerNames.next();
             if (headerName.startsWith("X-")) {
-                if(logger.isDebugEnabled()) {
-                    logger.debug("%%%%%%%%%%% Indetified customer header: " + headerName);
+                if (logger.isDebugEnabled()) {
+                    logger.debug("%%%%%%%%%%% Identified customer header: " + headerName);
                 }
                 parameters.add(new BasicNameValuePair(prefix + headerName, sipMessage.getHeader(headerName)));
+            } else if (headerName.startsWith("Diversion")) {
+
+                final String sipDiversionHeader = sipMessage.getHeader(headerName);
+                if (logger.isDebugEnabled()) {
+                    logger.debug("%%%%%%%%%%% Identified diversion header: " + sipDiversionHeader);
+                }
+                parameters.add(new BasicNameValuePair(prefix + headerName, sipDiversionHeader));
```

Comment:
@otsakir , I have use this nice feature where the custom headers are added with a prefix, in this case, the Header is "diversion", so that's the RVD variable